apps/contract-sync-job/src/ipfs.ts

```diff
@@ -0,0 +1,35 @@
+import fetch from 'node-fetch';
+
+const INFURA_SUBDOMAIN = process.env.INFURA_SUBDOMAIN;
+
+const getIpfsFromHash = (hash: string) => {
+  return `${INFURA_SUBDOMAIN}/ipfs/${hash.split('/').slice(2)}`;
+};
+
+export const fetchIPFS = async (
+  ipfsHash: string,
+): Promise<{
+  name: string;
+  details?: string;
+  proof?: string;
+}> => {
+  const auth =
+    'Basic ' +
+    Buffer.from(
+      process.env.INFURA_PROJECT_ID + ':' + process.env.INFURA_PROJECT_SECRET,
+    ).toString('base64');
+
+  const resp = await fetch(getIpfsFromHash(ipfsHash), {
+    headers: {
+      Authorization: auth,
```

Comment:
Also, I noticed things weren't working as expected with the custom domain. We needed auth and a different uri path